disperser/batcher/encoding_streamer.go

```diff
@@ -531,6 +531,10 @@ func (e *EncodingStreamer) CreateBatch() (*batch, error) {
 	metadatas := make([]*disperser.BlobMetadata, len(metadataByKey))
 	i := 0
 	for key := range metadataByKey {
+		err := e.transitionBlobToDispersing(ctx, metadataByKey[key])
```

Comment:
I kind of preferred having this in the batch handler since `CreateBatch` doesn't necessarily seem like it should be updating state. But doesn't seem important.

Reply:
Agree. But updating status for many blobs can have non-trivial latency, so it would be safer to update status before the method returns